bundledata.go

```diff
@@ -455,6 +459,41 @@ func (verifier *bundleDataVerifier) verifyRelations() {
 	}
 }
 
+func (verifier *bundleDataVerifier) verifyEndpointBindings() {
+	for name, svc := range verifier.bd.Services {
+		charm, ok := verifier.charms[name]
+		// Only thest the ok path here because the !ok path is tested in verifyServices
+		if ok {
+			for BoundInterface := range svc.EndpointBindings {
+				var ok bool
+
+				for _, relation := range charm.Meta().Provides {
+					if relation.Interface == BoundInterface {
+						ok = true
+						break
+					}
+				}
+
+				if !ok {
+					for _, relation := range charm.Meta().Requires {
+						if relation.Interface == BoundInterface {
+							ok = true
+							break
+						}
+					}
+				}
+
+				if !ok {
+					verifier.addErrorf(
+						"service %s wants to bind to interface %s, "+
+							"which isn't provided or required by charm",
```

Comment:
I think it will be helpful to include the space name in the error message.
s/to interface %s/endpoint %q to space %q/
s/isn't provided or required by charm/isn't among the defined charm relations/ ?